x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/ProfilingIndexManager.java

```diff
@@ -42,6 +42,7 @@
 class ProfilingIndexManager extends AbstractProfilingPersistenceManager<ProfilingIndexManager.ProfilingIndex> {
     // For testing
     public static final List<ProfilingIndex> PROFILING_INDICES = List.of(
+        ProfilingIndex.regular("profiling-costs", ProfilingIndexTemplateRegistry.PROFILING_COSTS_VERSION, OnVersionBump.KEEP_OLD),
```

Comment:
`KEEP_OLD` indicates that there will be some component that actively takes care of upgrades. E.g. for symbolization indices the symbolization service eventually will need to take care of migration an old index structure to the new one (not yet implemented). Do we also want that for `profiling-costs? Can we default to deleting it on version bumps as long as we control what is written to these indices?

Reply:
Good question. Currently, the data is generated/written manually (with a tool), so KEEP_OLD seems correct.
I assume this will change until 8.12 FF, and at that point we have to reconsider the flag.

Reply:
For now we can keep this as is. However, once we plan to bump the index version, we need to keep this in mind.

I'll approve the PR now because for the moment that's harmless.